perf_guide.md

```diff
@@ -67,17 +67,49 @@ addition is 2.7x slower than the one using vertical vector operations!
 
 ## Target-features
 
+Not all processors of a certain architecture will have SIMD processing units,
+and using a SIMD instruction when it's not supported will trigger a CPU exception.
+
+To allow building safe, portable programs, the Rust compiler will **not**, by default,
+generate any sort of vector instructions, unless it can statically determine
+they are supported (for example, on AMD64, SSE2 support is architecturally guaranteed).
+
 ### RUSTFLAGS
 
-TODO
+One of the easiest ways to benefit from SIMD is to allow the compiler
+to generate code using certain vector instruction extensions.
+
+The environment variable `RUSTFLAGS` can be used to pass options for code
+generation to the Rust compiler. These flags will affect **all** compiled crates.
+
+There are two flags which can be used to enable specific vector extensions:
+
+- `-C target-feature=<features>`: a comma-separated set of instruction extensions
+  to enable.
+  - **NOTE** (x86 specific): the `avx2` feature will _not_ enable `fma`, even though
+  all CPUs which support AVX2 also support FMA. You can enable this feature in addition to,
+  or separately from AVX2.
+
+<!-- TODO: list all target features for each platform -->
```

Comment:
I think this will be too much and a pain to keep up to date. Maybe we should mention how to obtain the target features available for each different target ? 

To list all available targets `rustc --print target-list`. 
To list all available target-features for a target: `rustc --target=${TRIPLE} --print target-features`